lib/pavilion/series.py

```diff
@@ -65,17 +236,18 @@ def __init__(self, pav_cfg, tests, _id=None):
                     "Could not get id or series directory in '{}': {}"
                     .format(series_path, err))
 
-            # Create a soft link to the test directory of each test in the
-            # series.
-            for test in tests:
-                link_path = utils.make_id_path(self.path, test.id)
+            if tests:
+                # Create a soft link to the test directory of each test in the
+                # series.
+                for test in tests:
+                    link_path = utils.make_id_path(self.path, test.id)
 
-                try:
-                    link_path.symlink_to(test.path)
-                except OSError as err:
-                    raise TestSeriesError(
-                        "Could not link test '{}' in series at '{}': {}"
-                        .format(test.path, link_path, err))
+                    try:
+                        link_path.symlink_to(test.path)
+                    except OSError as err:
+                        raise TestSeriesError(
+                            "Could not link test '{}' in series at '{}': {}"
+                            .format(test.path, link_path, err))
```

Comment:
This code is equivalent before and after.

Reply:
I have to create a TestSeries object without specifying tests (because I add the tests to it later)